src/hypothesis/searchstrategy/types.py

```diff
@@ -129,6 +135,19 @@ def from_typing_type(thing):
 for t in integer_types:
     _global_type_lookup[t] = st.integers()
 
+if hasattr(os, 'PathLike'):  # pragma: no branch
+    _global_type_lookup[os.PathLike] = \
+        st.fspaths(allow_pathlike=False).map(st._PathLike)
+
+# Also check for PY3 because there is a Python 2 pathlib port on pypi
+# which wont work
+if PY3 and pathlib is not None:  # pragma: no branch
+    _global_type_lookup[pathlib.Path] = \
```

Comment:
Right, the magical `builds()` fallback option kicked in.  Apart from that, all examples in `from_type(T)` should be instances of `T`, which means that we can resolve the parent type to a strategy for child types but not vice-versa.